### DISCUSSION TITLE
Modify Software for AutoFocus Raspberry Pi Zero Camera Operation

### DISCUSSION THREAD

**Discussion by GRBlake488** [alanbjohnston/CubeSatSim#137] [Q&A]
Greetings.

I came across a version of the Raspberry Pi Camera 1 with an AutoFocus lens. This requires software modification to enable.

Would someone be willing to modify the code to use it? I can see this would be a great benefit.

Thank You.

George Blake

https://www.amazon.com/Arducam-Autofocus-Raspberry-Motorized-Software/dp/B07SN8GYGD/ref=pd_ybh_a_5?_encoding=UTF8&psc=1&refRID=ATW5T84QPN9X5884FPT4

**Comment by alanbjohnston** [comment]
Hi George,

Thanks for your question. Some of what I have read about that camera seems to say that it is compatible. We use raspistill to take a photo. If that works then no changes are needed. Are you able to test this? For example, you could type this in a terminal window. 

`raspistill -v -o test.jpg`

If it outputs a test.jpg file, then we are good. 

Note that Raspberry Pi has messed up the camera apps and APIs with the new Bullseye software update, so please test this with Buster if you can. I am working on a software update that will be compatible with Bullseye. 

Good luck with your CubeSatSim!

Alan

**Reply by GRBlake488** [reply]
Alan, The question was not about whether or not the camera itself will work. The question was about adding the software function to control the focus. The advertisement says the focus is software controlled. This would have to be added in addition to the standard routine.

**Reply by alanbjohnston** [reply]
OK. How would this control work? 

Right now the only control input is a single push button to change modes, start, and shut down.  Everything else is set with a configuration file sim.cfg which is loaded as the CubeSatSim software starts up.

If you can give some more details of how this would work given the nature of the CubeSatSim.

**Reply by GRBlake488** [reply]
The Auto-Focus should work continuously, or at least, when you invoke SSTV mode. It should not need a mode of its own. Auto-focus is to refine the image of what you are taking. It is not Auto-Zoom. Just add to the existing code. If it is a fixed-focus camera as most are. If you add a variable Zoom lens, that is a different issue.

**Reply by GRBlake488** [reply]
BTW: Thanks for making the CubeSatSim. I know you can't fly it in space without ruggedizing it. I plan to use it for BalloonSat and also launch with a large Model Rocket into the upper atmosphere. 
I would have made a few things different. You could incorporate the CPU itself on a board using the Eagle files for Arduino or Raspberry Pi. Then add everything else. That would give you a good basis for a space-flight ready system.
I am building a two-axis Satellite tracker using a Pelco Pan / Tilt mount. You can see it at QRZ.com or LinkedIn.
73's.
W6BDD.

**Comment by GRBlake488** [comment]
I modified the software to stream video via the Wireless device on the board. I bought the RPZ2 that has this on-board. It appears to work fine along with the CubeSatSim software.

The next trick is to try the Auto-Focus code.
